bitsail-connectors/connector-druid/src/main/java/com/bytedance/bitsail/connector/druid/sink/DruidWriter.java

```diff
@@ -0,0 +1,267 @@
+   */
+  @Override
+  public List<CommitT> prepareCommit() {
+    return null;
+  }
+
+  private HttpURLConnection provideHttpURLConnection(final String coordinatorURL) throws IOException {
+    final URL url = new URL("http://" + coordinatorURL + DRUID_ENDPOINT);
+    final HttpURLConnection con = (HttpURLConnection) url.openConnection();
+    con.setRequestMethod("POST");
+    con.setRequestProperty("Content-Type", "application/json");
+    con.setRequestProperty("Accept", "application/json, text/plain, */*");
+    con.setDoOutput(true);
+    return con;
+  }
+
+  private ObjectMapper provideDruidSerializer() {
+    final ObjectMapper mapper = new ObjectMapper();
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    mapper.configure(MapperFeature.AUTO_DETECT_GETTERS, false);
+    mapper.configure(MapperFeature.AUTO_DETECT_FIELDS, false);
+    mapper.configure(MapperFeature.AUTO_DETECT_IS_GETTERS, false);
+    mapper.configure(MapperFeature.AUTO_DETECT_SETTERS, false);
+    mapper.configure(SerializationFeature.INDENT_OUTPUT, false);
+    mapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+    return mapper;
+  }
+
+  private DataSchema provideDruidDataSchema(final BitSailConfiguration writerConfig) {
+    final String datasource = writerConfig.getNecessaryOption(DATASOURCE, REQUIRED_VALUE);
+    final List<DimensionSchema> dimensionSchemas = this.columnInfos.stream()
+            .map(this::transformToDimensionSchema)
+            .collect(Collectors.toList());
+    return new DataSchema(
+            datasource,
+            new TimestampSpec(TIMESTAMP_SPEC_COLUMN_NAME, "auto", null),
+            new DimensionsSpec(dimensionSchemas),
+            null,
+            new UniformGranularitySpec(Granularities.HOUR, Granularities.MINUTE, false, null),
+            null
+    );
+  }
+
+  private DimensionSchema transformToDimensionSchema(final ColumnInfo columnInfo) {
+    final String columnName = columnInfo.getName();
+    final String columnType = columnInfo.getType().toLowerCase();
+    final TypeInfo<?> typeInfo = typeInfoConverter.fromTypeString(columnType);
+    final DimensionSchema dimensionSchema;
+    if (TypeInfos.STRING_TYPE_INFO.equals(typeInfo)) {
+      dimensionSchema = new StringDimensionSchema(columnName);
+    } else if (TypeInfos.FLOAT_TYPE_INFO.equals(typeInfo)) {
+      dimensionSchema = new FloatDimensionSchema(columnName);
+    } else if (TypeInfos.DOUBLE_TYPE_INFO.equals(typeInfo)) {
+      dimensionSchema = new DoubleDimensionSchema(columnName);
+    } else if (TypeInfos.LONG_TYPE_INFO.equals(typeInfo)) {
+      dimensionSchema = new LongDimensionSchema(columnName);
+    } else {
+      throw new BitSailException(UNSUPPORTED_OPERATION,
+              "Column type name " + columnName + " type " + columnType + " is not supported.");
+    }
+    return dimensionSchema;
+  }
+
+  @VisibleForTesting
+  ParallelIndexIOConfig provideDruidIOConfig(final StringBuffer data) {
```

Comment:
brief explanation about what this do would be helpful.

Reply:
Sure will add the following javadoc in next commit.
```
Provide ioConfig that influences how data is read from a source system. Here we generate a CSV file from writing
data and use the Native batch Ingestion method (https://druid.apache.org/docs/latest/ingestion/index.html#batch)
to load from CSV file. More details in https://druid.apache.org/docs/latest/ingestion/ingestion-spec.html#ioconfig.
```